thorlcr/graph/thgraphslave.cpp

```diff
@@ -111,6 +111,29 @@ class CBarrierSlave : public CInterface, implements IBarrier
     virtual const mptag_t queryTag() const { return tag; }
 };
 
+
+
+bool isFastThrough(IThorDataLink *input)
+{
+    CSlaveActivity *act = (CSlaveActivity *)input->queryFromActivity();
+    if (act)
+    {
+        ThorDataLinkMetaInfo info;
+        act->getMetaInfo(info);
```

Comment:
Slight danger of O(N^2) processing if each activity also calls down to its input's getMetaInfo().  I doubt it needs addressing, but potentially could become an issue.  If an issue it could always cache the meta in the activity, and reset it at the start of each graph execution.  (It may solve potential disk read issues as well.)